File: backend/app/models/technician_service.py
from uuid import UUID
from pydantic import BaseModel
from typing import Dict, Any
from pydantic import BaseModel, Field


class BaseTechnicianService(BaseModel):
    technician_id: UUID
    service_id: UUID
    experience_years: int
    price: float = Field(gt=0)


class TechnicianServiceInDB(BaseTechnicianService): ...


class TechnicianServiceCreate(BaseTechnicianService):
    class Config:
        json_schema_extra: Dict[str, Any] = {
            "example": {
                "technician_id": "123e4567-e89b-12d3-a456-426655440000",
                "service_id": "123e4567-e89b-12d3-a456-426655440000",
                "experience_years": 2,
                "price": 150.00
            }
        }


class TechnicianServiceUpdate(BaseModel):
    experience_years: int


class TechnicianServiceResponse(BaseTechnicianService):
    class Config:
        json_schema_extra: Dict[str, Any] = {
            "example": {
                "technician_id": "123e4567-e89b-12d3-a456-426655440000",
                "service_id": "123e4567-e89b-12d3-a456-426655440000",
                "experience_years": 2,
                "price": 150.00
            }
        }

File: frontend/src/lib/types/technician_service.ts
import { z } from "zod";

const BaseTechnicianServiceSchema = z.object({
    technician_id: z.string().uuid(),
    service_id: z.string().uuid(),
    experience_years: z.number().min(0),
    price: z.number().gt(0)
});

export const TechnicianServiceCreateSchema = BaseTechnicianServiceSchema.extend({});

export const TechnicianServiceUpdateSchema = z.object({
    experience_years: z.number().min(0).optional(),
    price: z.number().gt(0).optional()
});

export const TechnicianServiceResponseSchema = BaseTechnicianServiceSchema.extend({
    id: z.string().uuid(),
});

export type TechnicianServiceCreate = z.infer<typeof TechnicianServiceCreateSchema>;
export type TechnicianServiceUpdate = z.infer<typeof TechnicianServiceUpdateSchema>;
export type TechnicianServiceResponse = z.infer<typeof TechnicianServiceResponseSchema>;
